default.nix

```diff
@@ -1,27 +1,26 @@
 # Optionally put the specified nix version of the package in the environment
-{ environment ? "default"
-}:
+{ environment ? "shell" }:
 let
-  pkgs = import ./nixpkgs-pinned.nix {};
+  pkgs = import ./nix/nixpkgs-pinned.nix { };
 
-  defaultEnv = pkgs.buildEnv {
-    name = "hoff-devenv";
-    paths = [
+  defaultEnv = pkgs.haskellPackages.shellFor {
+    packages = p: [ p.hoff ];
+
+    buildInputs = [
       pkgs.dia
       pkgs.dpkg
       pkgs.git
-      pkgs.haskellPackages.haskell-language-server
-      pkgs.haskellPackages.ghc # Needed for the language server
-      pkgs.haskellPackages.stylish-haskell
       pkgs.niv
       pkgs.shellcheck
-      pkgs.stack
+
+      pkgs.haskellPackages.cabal-install
+      pkgs.haskellPackages.haskell-language-server
+      pkgs.haskellPackages.implicit-hie
```

Comment:
I feel like we have a love-hate relationship with the `gen-hie` command since we have tried to get rid of it on other places, I'm not sure. I tried running everything from a fresh clone without it and it still worked o.o

Reply:
Yeah it's only needed to generate the `hie.yaml` file once. It usually does the right thing and I used it to generate a new one for Cabal, though I did have to manually add the entries for the end to end tests since they live in the same directory as the other tests.

Reply:
I'm guessing that since we currently have one, we won't need to run it ever unless there's a big enough change in the module hierarchy, which I don't think is going to happen anytime soon. If I were you I'd remove it out of sheer "we don't need this right now or next month", but we've had this happen elsewhere with `niv` and `cachix` too so I'll trust your judgement.

Reply:
It's indeed not really needed once set up, but our other repos all have it so I added it here too. In the name of consistency.